### ISSUE TITLE
ASR Quartznet Fine Tuning with additional datasets

### ISSUE THREAD

**Issue by anujsuchchal** [NVIDIA/NeMo#1215]
Hi,

Can somebody please help me here? The pre-trained model 'Multidataset-QuartzNet15x5' is giving around 60% accuracy on my domain test data. I know it is trained on 6 different datasets. So I downloaded more open-source data (VoxForge) and trying to fine tune it using small learning rate (around 0.0001)
Pretrained model has WER of 10.05% on librispeech dev-other and I am using same validation data in fine tuning to compare performance.
Now, the issue is, even when model is learning fine with this additional data (train loss and train WER is going down), the overall validation loss on dev-other is increasing (~1% jump every epoch). It looks like whenever I train on additional data, pretrained model tends to 'skew' towards this new data.
So I have found that training will all 7 datasets from scratch is not similar to training with 6 datasets from scratch and then fine tuning with 7th dataset. Correct me if I am wrong.
So do I need to download all 6 datasets on which pretrained model was trained and add it up with my additional data and train from scratch with all data? Sorry but I am only having single Nvidia Quadro RTX 6000 and it will take months on it.
Am I doing anything wrong? Do I need to tune any other hyperparameter apart from learning rate?
Below are the hyperparameters I am currently using:-

1. Learning Rate - 0.0001 (Found this one optimum as train loss was going up with any other value greater than this)
2. Optimizer - Novograd
3. Warmup Ratio - 0.02
4. Batch Size - 32
5. Beta1 - 0.95
6. Beta2 - 0.5
7. Weight Decay - 0.001
8. LR Policy - CosineAnnealing

My aim is to increase accuracy on my domain data (I don't have data to train from my domain FYI). Please guide.

**Comment by okuchaiev** [comment]
>"So I have found that training will all 7 datasets from scratch is not similar to training with 6 datasets from scratch and then fine tuning with 7th dataset"
- You are correct. This is not the same.

>"My aim is to increase accuracy on my domain data"
- So the best way to do it is to fine-tune on a data from your domain. I would not worry about perf on librispeech while fine-tuning, it is expected to drop a little. Important metric to monitor is perf on your data. Is VoxForge more similar to your data? If yes, then fine-tune with it.
- There is one more trick you can try but it is currently not supported properly. You can try training an external language model on your data (you only need text data for that). And then use that LM as part of beam search decoder.
Like I said this is not properly supported yet but we are working on it. This might help: https://github.com/NVIDIA/NeMo/issues/1156

**Comment by anujsuchchal** [comment]
@okuchaiev Thanks for your quick response.. I am trying both ways you suggested. Also, Can you please confirm If I am using correct hyperparameters or not? Do I need to change LR only or try tuning any other value as well?
